flight/src/types.rs

```diff
@@ -0,0 +1,102 @@
+use std::fmt;
+
+use cid::Cid;
+use serde::{Deserialize, Serialize};
+use serde_bytes::ByteBuf;
```

Comment:
Why do we need `serde` for these types? Can we use a simple Vec<u8> instead of the ByteBuf since we are serializing or deserializing these types?